src/lightning/__version__.py

```diff
@@ -1 +1,14 @@
-version = "1.8.0rc1"
+# in case you have not installed the package and have only set python path to this source
+
+import subprocess
+
+
+def get_git_revision_short_hash() -> str:
+    # https://stackoverflow.com/a/21901260/4521646
+    return subprocess.check_output(["git", "rev-parse", "--short", "HEAD"]).decode("ascii").strip()
```

Comment:
I'm not a fan of this solution.

This creates changes to the filesystem (`os.remove` and `os.copy`). And it creates a dependency on `git`, which is not always available. Then you'd get an empty version string which would be very confusing.

I strongly suggest that we keep the different static version files as before.

And to ensure they don't diverge, add a test in CI where we import all packages and assert that the version numbers match.

Reply:
I agree, this is the reason tools like `setuptools-scm` etc. are not more widely adopted.

Reply:
So you suggest that you need to remember that all sub-packages need to be updated consistently and if not release process will fail and all need to be edited and released manually because some were uploaded...
I feel that this is a bigger burden, and we have the edit in the codebase already, so this does not introduce new; it just improves it

Reply:
> And it creates a dependency on git, which is not always available. Then you'd get an empty version string which would be very confusing.

If you use any way of installation, you get the semantic version, the only time you rely on git is if your IDE has set the source to `src/` folder and in such case, you have cloned this repository (the only other way would download repo as an archive and unzip it, then you get "")

Reply:
> to ensure they don't diverge, add a test in CI where we import all packages and assert that the version strings match.

yes, that would be an option, but as mentioned in https://github.com/Lightning-AI/lightning/pull/15374#discussion_r1008001443 we have such edits due to the installation already anyway :/

Reply:
Which edits? `set_version_today` is not used anywhere

Reply:
> If you use any way of installation, you get the semantic version, the only time you rely on git is if your IDE has set the source to src/ folder and in such case, you have cloned this repository (the only other way would download repo as an archive and unzip it, then you get "")

But this is inconsistent and hard to test as well. E.g. the `checkout` action on GH does not preserve the `.git` folder and thereby makes it hard to actually test this. (See https://github.com/actions/checkout/issues/335 )

Reply:
> `set_version_today` is not used anywhere

I don't it anywhere, it was used for `lightning` package until we unified to semver

Reply:
> But this is inconsistent and hard to test as well. E.g. the `checkout` action on GH does not preserve the `.git` folder and thereby makes it hard to actually test this.

that is fine, we run install in CI